Tabby/types/api_request.ts

```diff
@@ -0,0 +1,160 @@
+import { Non200Status } from "./error";
+
+/**
+ * Datatype for the http helper function
+ * 
+ * @param {string} domain The URL for the HTTP call.
+ * @param {string} route Optional. Follows after the URL/... 
+ * @param {string} body Optional. Build the body for the request
+ * @param {} method Specify what method to use. Different methods
+ *  may require parameters be utilized.
+ * @param {string} type Specify the datatype of the body
+ * 
+ * @note For continuity, domain is expected to contain the /
+ */
+type http = {
+    domain: string,     // URL
+    route?: string,     // What is the target route
+    body?: string,      // Body of API request
+    method: 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE',   // HTTP Method
+    type: string        // 'application/json'
+}
+
+/*
+    /// HTTP Request: https://bobbyhadz.com/blog/typescript-http-request
+    /// Exception handling: https://www.youtube.com/watch?v=AdmGHwvgaVs
+
+    /// NOTE: For all following request calls, follow the example of
+    /// the HTTP GET Example for how best to use the two functions
+
+    // Imports
+    import { http_callback } from '@/types/api_request';
+    import { catchErrorTyped } from '@/types/error_handle';
+    import { Non200Status } from '@/types/error';
+
+    // HTTP GET Example
+    const [error, value] = await catchErrorTyped(http_callback({
+        domain: "https://reqres.in/api/",
+        route: "members",
+        method: "GET",
+        type: "application/json"
+    }), [Non200Status]);
+    // ^^^ Will only catch one type of error.
+
+    if (error) {
+        console.log("Error Found ", error.message);
+    } else {
+        console.log(value);
+    }
+
+    // HTTP POST Example
+    await http_callback({
+        domain: "https://reqres.in/api/",
+        route: 'users',
+        method: "POST",
+        body: JSON.stringify({
+            name: 'John Deering',
+            job: 'manager',
+        }),
+        type: "application/json"
+    });
+
+    // HTTP PUT Example
+    await http_callback({
+        domain: "https://reqres.in/api/",
+        route: 'users/2',
+        method: "PUT",
+        body: JSON.stringify({
+            name: 'morpheus',
+            job: 'Zion Resident',
+        }),
+        type: "application/json"
+    });
+
+    // HTTP PATCH Example
+    await http_callback({
+        domain: "https://reqres.in/api/",
+        route: 'users/2',
+        method: "PATCH",
+        body: JSON.stringify({
+            name: 'Zion Resident',
+            job: 'Morphing',
+        }),
+        type: "application/json"
+    });
+
+    // HTTP DELETE Example
+    await http_callback({
+        domain: "https://reqres.in/api/",
+        route: 'users/2',
+        method: "DELETE",
+        type: "application/json"
+    });
+*/
```

Comment:
Why is this block commented out?

I don't really have an issue with it, but I would like to know if there is a reason on why.